### PR TITLE
Pull request for adding modifiable parameters to the CSM

### PR DESCRIPTION
**Pull request by MarjorieH**

### CODE REVIEW

**Comment by jlaura** [comment]
@twilson271828 It looks like your PR added this functionality?  Can you take a look at this PR and confirm that we should not merge?  Or am I missing something that this PR address something different.  For example, checkout lines (added) 140 - 142.  This looks like it should follow your implementation instead.

**Comment by twilson271828** [comment]
That's true, although my PR did not make any changes to the MdisPlugin.cpp class.  Also,
there are changes to MdisNacSensorModel::getSensorPosition which I did not include in my PR.
I would recommend not merging these changes.  I will close it, and then we can come back to it again in a future sprint if needed.

**Comment by MarjorieH** [comment]
The big change in this pull request is that o,p,k and sensor position x,y,z are replaced by the currentParameterValue array. This is so that someone can use setParameterValue in order to change these values based on where the sensor is and where it is pointing. Currently, if the user has a sensor model initialized, then changes the the parameter values, and attempts to use g2i or i2g, these functions will run based on the old parameter values, because these functions are not using the currentParameterValue array.

File: src/objs/MdisNacSensorModel.cpp [hunk 1]
```diff
@@ -132,60 +136,32 @@ csm::ImageCoord MdisNacSensorModel::groundToImage(const csm::EcefCoord &groundPt
...
+  zl = m_currentParameterValue[2];
 
   double x, y, z;
   x = groundPt.x;
   y = groundPt.y;
   z = groundPt.z;
 
   double xo, yo, zo;
-
-  xo = xl - x - getValue(0,adjustments);
```
Comment (MarjorieH) [inline src/objs/MdisNacSensorModel.cpp]: keep getValue

File: src/objs/MdisNacSensorModel.cpp [hunk 2]
```diff
@@ -132,60 +136,32 @@ csm::ImageCoord MdisNacSensorModel::groundToImage(const csm::EcefCoord &groundPt
...
-  yo = yl - y - getValue(1,adjustments);
-  zo = zl - z - getValue(2,adjustments);
```
Comment (MarjorieH) [inline src/objs/MdisNacSensorModel.cpp]: keep getValue() for adjustments

File: src/objs/MdisNacSensorModel.cpp [hunk 3]
```diff
@@ -132,60 +136,32 @@ csm::ImageCoord MdisNacSensorModel::groundToImage(const csm::EcefCoord &groundPt
...
+  yo = yl - y;
+  zo = zl - z;
 
   double f;
   f = m_focalLength;
 
   // Camera rotation matrix
   double m[3][3];
-  //
-  //calcRotationMatrix(m,adjustments);
```
Comment (MarjorieH) [inline src/objs/MdisNacSensorModel.cpp]: uncomment calcRotationMatrix(m,adjustments);, delete calcRotationMatrix(m);

File: src/objs/MdisNacSensorModel.cpp [hunk 4]
```diff
@@ -132,60 +136,32 @@ csm::ImageCoord MdisNacSensorModel::groundToImage(const csm::EcefCoord &groundPt
...
-csm::ImageCoord MdisNacSensorModel::groundToImage(
-   const csm::EcefCoord&      groundPt,
-   const std::vector<double>& adjustments,
-   double                     desired_precision,
-   double*                    achieved_precision,
-   csm::WarningList*          warnings ) const {
-
-
-
   double xl, yl, zl;
```
Comment (MarjorieH) [inline src/objs/MdisNacSensorModel.cpp]: Do not delete

File: src/objs/MdisNacSensorModel.cpp [hunk 5]
```diff
@@ -105,13 +107,15 @@ MdisNacSensorModel::MdisNacSensorModel() {
...
-
-  m_currentParameterValue.assign(m_nParameters,0.0);
-  m_currentParameterCovariance.assign(m_nParameters*m_nParameters,0.0);
-  m_noAdjustments.assign(m_nParameters,0.0);
+  
+  // Initialize parameter values
+  m_currentParameterValue = new double[m_numParameters];
+  for (int i = 0; i < m_numParameters; i++) {
+    m_currentParameterValue[i] = 0.0;
+  }
```
Comment (MarjorieH) [inline src/objs/MdisNacSensorModel.cpp]: assign values instead of using forloop

Comment (thareUSGS) [inline src/objs/MdisNacSensorModel.cpp]: +1

**Pull request merged**